### PR TITLE
JRMP2-4_Finished_working_on_the_story

### PR DESCRIPTION
**Pull request by dmytro-khyzhniak**
I have prepared the foundations on which the system will work in the future [BE].

- Ability to work with TypeScript.
- A basic app, router, controller, model.
- A basic imitation of the database and the ability to work with it.

### CODE REVIEW

File: backend/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true,
+    "outDir": "./prod/"
+  },
+  "files": [
+    "./dev/app.ts",
+    "./dev/controller.ts",
+    "./dev/model.ts",
+    "./dev/router.ts",
+  ]
```
Comment (DmitryKorolDevPro) [inline backend/tsconfig.json]: if we add any new file will it need to be added to this list? can you try to make it more versatile? (for example, let's process ALL files except ...)

File: backend/dev/app.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const express = require('express');
+const router = require('./router.js');
```
Comment (DmitryKorolDevPro) [inline backend/dev/app.ts]: "imports" syntax  - more native for typescript

File: backend/dev/app.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+const app = express();
+const port:number = 4200;
```
Comment (DmitryKorolDevPro) [inline backend/dev/app.ts]: it is good practice to pass the port through environment variables. (you can google what environment variables are)

good practice two is to describe the default value if the environment variable is not set

File: backend/dev/controller.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Controller {
+  constructor() {
+    this.test();  
+  }
+
+  test() {
+    console.log('+controller');
+  }
```
Comment (DmitryKorolDevPro) [inline backend/dev/controller.ts]: I see no point in a separate method. it is better to make a more explicit message to the console log. something like "the controller is alive!"

File: backend/dev/controller.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+}
+
+module.exports = new Controller();
```
Comment (DmitryKorolDevPro) [inline backend/dev/controller.ts]: if you work with a typescript - you can export without a module

File: backend/dev/model.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      console.log('+database');
+    } catch (error) {
+      console.log(error);
+      console.log('-database');
+    }
+  }
+}
+
+module.exports = new Model();
+
```
Comment (DmitryKorolDevPro) [inline backend/dev/model.ts]: comments on previous files also apply to this.

don't leave so many extra spaces.

File: backend/prod/app.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+"use strict";
+const express = require('express');
+const router = require('./router.js');
+const app = express();
+const port = 4200;
+app.use('/', router);
+app.listen(port, () => {
+    console.log(`Example app listening at http://localhost:${port}`);
+});
```
Comment (DmitryKorolDevPro) [inline backend/prod/app.js]: the build results should not be under the git. 
under the git, only the files in which we write the code . 
files generated by the machine are not.

File: backend/dev/app.ts (hunk 3 above)
Comment (dmytro-khyzhniak) [reply backend/dev/app.ts]: Yeah, I have already used it in our last project. Thanks for reminding me :)
Changed to:
`let port:number = process.env.PORT === undefined ? 4200 : +process.env.PORT;`

File: backend/tsconfig.json (hunk 1 above)
Comment (dmytro-khyzhniak) [reply backend/tsconfig.json]: Now I'm using:
` "include": [
    "./dev/*.ts"
  ]`

File: backend/dev/app.ts (hunk 2 above)
Comment (dmytro-khyzhniak) [reply backend/dev/app.ts]: Changed to:
`import express from "express";
import { router } from './router.js';`

File: backend/dev/model.ts (hunk 6 above)
Comment (dmytro-khyzhniak) [reply backend/dev/model.ts]: My bad.

File: backend/dev/app.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import express from "express";
+import { router } from './router.js';
+
+const app = express();
+let port:number = process.env.PORT === undefined ? 4200 : +process.env.PORT;
```
Comment (DmitryKorolDevPro) [inline backend/dev/app.ts]: it might be easier

process.env.PORT || 4200

why let? are we planning to change the variable?

**Review by DmitryKorolDevPro — APPROVED** [review]

**Pull request merged**